app.rb

```diff
@@ -0,0 +1,167 @@
+      puts 'Failed to create a person'
+    end
+
+    student
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    if age.positive? && name.strip.length.positive?
+      teacher = Teacher.new(age, name, specialization)
+      if teacher
+        puts 'Person created successfully'
+        teacher
+      else
+        puts 'Failed to create a person'
+      end
+    else
+      puts 'Failed to create a person'
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    if title.strip.size.positive? && author.strip.size.positive?
+      book = Book.new(title, author)
+      if book
+        puts 'Book created successfully'
+        @books << book
+      else
+        puts 'Failed to create a book'
+      end
+    else
+      puts 'Failed to create a book'
+    end
+  end
+
+  def create_rent
+    puts 'Select a book from the following list by number'
+    @books.each_with_index { |b, i| puts "#{i}) Title: #{b.title}, Author: #{b.author}" }
+    book = @books[gets.chomp.to_i]
+
+    puts 'Select a person from the following list by number (not id)'
+    @people.each_with_index { |per, i| puts "#{i}) [#{per.type}] Name: #{per.name}, ID: #{per.id}, Age: #{per.age}" }
+    person = @people[gets.chomp.to_i]
+
+    print 'Date: '
+    date = gets.chomp
+    if date.strip.size.positive?
+      rent = Rental.new(date, book, person)
+      if rent
+        puts 'Rental created successfully'
+        @rents << rent
+      else
+        puts 'Failed to create a rent'
+      end
+    else
+      puts 'Failed to create a rent'
+    end
+  end
+
+  def student_or_teacher
+    print 'Do u wanna create a student(1) or teacher(2)? [Input the number]: '
+    opt3 = gets.chomp.to_i
+    case opt3
+    when 1
+      student = create_student
+      @people << student if student
+    when 2
+      teacher = create_teacher
+      @people << teacher if teacher
+    end
+  end
+
+  def list_rentals
+    print 'ID of person: '
+    id = gets.chomp.to_i
+    @rents.each do |rent|
+      puts "Date: #{rent.date}, Book \"#{rent.book.title}\" by \"#{rent.book.author}\"" if rent.person.id == id
+    end
+  end
+
+  def start
+    puts
+    puts 'Please choose an option by enter a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person ID'
+    puts '7 - Exit'
+
+    gets.chomp.to_i
+  end
+
+  def run # rubocop:disable Metrics/CyclomaticComplexity
+    100.times do
+      opt = start
+
+      case opt
+      when 1
+        books.each { |b| puts "Title: \"#{b.title}\"; Author: \"#{b.author}\"" }
+      when 2
+        people.each { |per| puts "[#{per.type}] Name: #{per.name}, ID: #{per.id}, Age: #{per.age} " }
```

Comment:
- [ ] I'm encountering a bug when I try to list the people that were created and it's originating from the code at line `152` and when I try to create a rental as well which comes from line `91` and the main issue is because you're trying to access an undefined property called `type` in your code so I'd suggest you fix this to ensure the proper functionality of your code.
- This is the error shown
  ![image](https://github.com/leloufadel/OOP-School-Library/assets/104968236/23b178c8-de56-433a-bc82-16dc4717c626)
  ![image](https://github.com/leloufadel/OOP-School-Library/assets/104968236/6ae1b80d-390b-4e61-8cd4-cec221f64ca7)